app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java

```diff
@@ -225,6 +254,37 @@ private void onEnterEntryResult(int resultCode, Intent data) {
         }
     }
 
+    private void onScanImageResult(int resultCode, Intent intent) {
+        if (resultCode == RESULT_OK) {
+            Uri inputFile = (intent.getData());
+            InputStream inputStream;
+            Bitmap bitmap;
+
+            try {
+                inputStream = getContentResolver().openInputStream(inputFile);
```

Comment:
This InputStream is never closed. Please do so, preferably by turning this into a try-with-resources statement.